hibernate-core/src/main/java/org/hibernate/engine/jdbc/connections/spi/DatabaseConnectionInfo.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or http://www.gnu.org/licenses/lgpl-2.1.html
+ */
+package org.hibernate.engine.jdbc.connections.spi;
+
+import org.hibernate.dialect.DatabaseVersion;
+import org.hibernate.dialect.SimpleDatabaseVersion;
+
+/**
+ * @author Jan Schatteman
+ */
+public interface DatabaseConnectionInfo {
+	public static final String NOT_PROVIDED = "Not Provided";
```

Comment:
@jrenaat I'm not sure it's better to have a magical string instead of just `null`.

The code which actually logs can easily render `null` -> `"Not provided"`

WDYT?

Reply:
That is true, but I just viscerally detest using null

Reply:
Dunno, to me it's not about which is better/worse, I just thought it was better to return default values consistently for everything such that I don't have to pollute caller code with nullchecks. if i return null for those strings, then i could/should also return null for the version.
I mean if this is a big deal for you then yeah, i'll change it, but you'll not get me to *like* it ;)

Reply:
It's not a big deal at all. We can always change it later if needed.